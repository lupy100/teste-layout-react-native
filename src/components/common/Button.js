// @flow
import * as React from 'react';
import styled from 'styled-components/native';

type ButtonProps = {
  isGreen?: boolean,
};

const Button = styled.TouchableOpacity`
  padding: 15px 25px 15px 25px;
  border-radius: 50;
  background-color: ${(props: ButtonProps) =>
    props.isGreen ? 'rgba(106,227,104, 0.8)' : 'rgba(0, 0, 0, 0.4)'};
`;
const ButtonText = styled.Text`
  font-size: 20;
  font-weight: bold;
  color: #ffffff;
`;
type Props = {
  text?: string,
  isGreen?: boolean,
  children?: React.Node,
};

export default ({ text, isGreen, children, ...props }: Props) => (
  <Button isGreen={isGreen} {...props}>
    {text ? <ButtonText>{text}</ButtonText> : children}
  </Button>
);
