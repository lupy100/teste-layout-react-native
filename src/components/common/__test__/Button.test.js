// @flow
import * as React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import Button from '../Button';

describe('<Button />', () => {
  it('renders Button with text', () => {
    const tree = renderer.create(<Button text="Button text" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Button with text and isGreen', () => {
    const tree = renderer
      .create(<Button text="Button text" isGreen />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Button with children', () => {
    const tree = renderer
      .create(
        <Button>
          <Text>This is the children text</Text>
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
