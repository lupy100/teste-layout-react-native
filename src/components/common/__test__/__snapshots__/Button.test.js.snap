// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders Button with children 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.4)",
      "borderRadius": 50,
      "opacity": 1,
      "paddingBottom": 15,
      "paddingLeft": 25,
      "paddingRight": 25,
      "paddingTop": 15,
    }
  }
>
  <Text>
    This is the children text
  </Text>
</View>
`;

exports[`<Button /> renders Button with text 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.4)",
      "borderRadius": 50,
      "opacity": 1,
      "paddingBottom": 15,
      "paddingLeft": 25,
      "paddingRight": 25,
      "paddingTop": 15,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 20,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Button text
  </Text>
</View>
`;

exports[`<Button /> renders Button with text and isGreen 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(106,227,104, 0.8)",
      "borderRadius": 50,
      "opacity": 1,
      "paddingBottom": 15,
      "paddingLeft": 25,
      "paddingRight": 25,
      "paddingTop": 15,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 20,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Button text
  </Text>
</View>
`;
